# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build gcc-g++-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

if PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists sdl-build; then
  SDL=1
  REQUIRES="$REQUIRES sdl-build"
fi

# configure (ac-2.71; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/dfb/lib/pkgconfig ./configure --disable-sherlock265 --disable-static --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/bin
install acceleration-speed/.libs/acceleration_speed $DESTDIR/dfb/bin
test $SDL && install dec265/.libs/dec265 $DESTDIR/dfb/bin
install tools/.libs/bjoentegaard $DESTDIR/dfb/bin
install tools/.libs/block-rate-estim $DESTDIR/dfb/bin
install tools/.libs/gen-enc-table $DESTDIR/dfb/bin
install tools/.libs/rd-curves $DESTDIR/dfb/bin
install tools/.libs/yuv-distortion $DESTDIR/dfb/bin
install -d $DESTDIR/dfb/include/libde265
install -m 644 libde265/de265.h $DESTDIR/dfb/include/libde265
install -m 644 libde265/de265-version.h $DESTDIR/dfb/include/libde265
install -d $DESTDIR/dfb/lib
install libde265/.libs/libde265.so.0.* $DESTDIR/dfb/lib/libde265.so.0
ln -sf libde265.so.0 $DESTDIR/dfb/lib/libde265.so
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 libde265.pc $DESTDIR/dfb/lib/pkgconfig

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/libde265-build.pc << EOF
Name: libde265
Version: 1.0.11
Description: H.265/HEVC decoder
Requires: $REQUIRES

devel=\\
/dfb/include/libde265/de265.h \\
/dfb/include/libde265/de265-version.h \\
/dfb/lib/libde265.so \\
/dfb/lib/pkgconfig/libde265.pc

exec=\\
/dfb/bin/acceleration_speed \\
/dfb/bin/bjoentegaard \\
/dfb/bin/block-rate-estim \\
EOF
test $SDL && cat >> $DESTDIR/dfb/share/pkgconfig/libde265-build.pc << EOF
/dfb/bin/dec265 \\
EOF
cat >> $DESTDIR/dfb/share/pkgconfig/libde265-build.pc << EOF
/dfb/bin/gen-enc-table \\
/dfb/bin/rd-curves \\
/dfb/bin/yuv-distortion \\
/dfb/lib/libde265.so.0
EOF

exit
--- libde265-1.0.11.orig/configure.ac
+++ libde265-1.0.11/configure.ac
@@ -21,8 +21,6 @@
 
 # ---------------------------------------------------------------------------
 
-AC_CANONICAL_SYSTEM
-
 AC_SUBST(LIBDE265_CURRENT)
 AC_SUBST(LIBDE265_REVISION)
 AC_SUBST(LIBDE265_AGE)
@@ -109,9 +107,9 @@
 # Check if "__STRICT_ANSI__" is required.
 AC_MSG_CHECKING([if __STRICT_ANSI__ is required])
 AC_LANG_PUSH(C++)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 #include <vector>
-],[],[need_strict_ansi=no],[need_strict_ansi=yes]);
+],[])],[need_strict_ansi=no],[need_strict_ansi=yes]);
 AC_LANG_POP(C++)
 if eval "test x$need_strict_ansi = xyes"; then
   CFLAGS="$CFLAGS -D__STRICT_ANSI__"
@@ -122,21 +120,21 @@
 # Check if "std::shared_ptr" is "std::tr1::shared_ptr"
 AC_MSG_CHECKING([for std::shared_ptr])
 AC_LANG_PUSH(C++)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 #include <memory>
 ],[
 class A {};
 std::shared_ptr<A> a;
-],[has_std_shared_ptr=yes],[has_std_shared_ptr=no]);
+])],[has_std_shared_ptr=yes],[has_std_shared_ptr=no]);
 AC_MSG_RESULT([$has_std_shared_ptr])
 if eval "test x$has_std_shared_ptr = xno"; then
   AC_MSG_CHECKING([for std::tr1::shared_ptr])
-  AC_TRY_COMPILE([
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
   #include <tr1/memory>
   ],[
   class A {};
   std::tr1::shared_ptr<A> a;
-  ],[has_std_tr1_shared_ptr=yes],[has_std_tr1_shared_ptr=no]);
+  ])],[has_std_tr1_shared_ptr=yes],[has_std_tr1_shared_ptr=no]);
   AC_MSG_RESULT([$has_std_tr1_shared_ptr])
   if eval "test x$has_std_tr1_shared_ptr = xyes"; then
     AC_DEFINE(USE_STD_TR1_NAMESPACE,1,[Define to 1 if the std::tr1 namespace should be included in the std namespace.])
@@ -147,7 +145,7 @@
 # Check if "std::move" is available (assume always available with clang)
 AC_MSG_CHECKING([for std::move])
 AC_LANG_PUSH(C++)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 #include <utility>
 ],[
 #if !defined(__clang__)
@@ -155,7 +153,7 @@
 A* a = new A();
 A* b = std::move(a);
 #endif
-],[has_std_move=yes],[has_std_move=no]);
+])],[has_std_move=yes],[has_std_move=no]);
 AC_LANG_POP(C++)
 if eval "test x$has_std_move = xno"; then
   AC_DEFINE(NEED_STD_MOVE_FALLBACK,1,[Define to 1 if a fallback for "std::move" is required.])
@@ -165,11 +163,11 @@
 # Check if "nullptr" is available
 AC_MSG_CHECKING([for nullptr])
 AC_LANG_PUSH(C++)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 ],[
 class A {};
 A* a = nullptr;
-],[has_nullptr=yes],[has_nullptr=no]);
+])],[has_nullptr=yes],[has_nullptr=no]);
 AC_LANG_POP(C++)
 if eval "test x$has_nullptr = xno"; then
   AC_DEFINE(NEED_NULLPTR_FALLBACK,1,[Define to 1 if a fallback for "nullptr" is required.])
